api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -660,19 +660,12 @@ public void changePassword(User user, String oldPassword, String newPassword) th
 		updatePassword(user, newPassword);
 	}
 	
-	/**
-	 * This is for internal use only. DO NOT CALL THIS METHOD.
-	 * 
-	 * @param user The user's password to change
-	 * @param newPassword The password to change it to
-	 */
-	@Authorized(PrivilegeConstants.EDIT_USER_PASSWORDS)
+	@Override
 	public void changePassword(User user, String newPassword) {
-		if (!Daemon.isDaemonThread() || !Context.getUserContext().getAuthenticatedUser().isSuperUser()) {
```

Comment:
I'm not sure why this was added, but from looking at the logic, it would break existing functionality, as it only allows access if the user is a system developer.

From a search, I believe this method was actually being used anywhere, but I could be missing something?

Reply:
I agree.  There is already a privilege check around this.  Why is this additional check needed?  If we think most users should have the EDIT_USER_PASSWORDS privileged removed, that's reasonable, but we shouldn't restrict editing users to System Developer role specifically.